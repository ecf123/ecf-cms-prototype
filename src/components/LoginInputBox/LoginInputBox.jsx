import React from 'react'
import './LoginInputBox.scss'

const LoginInputBox = ({labelText}) => {
  return (
    <div>
        <label htmlFor="inputBox">{labelText}</label>
        <input className="inputBox" type="text" />
    </div>
  )
}

export default LoginInputBox