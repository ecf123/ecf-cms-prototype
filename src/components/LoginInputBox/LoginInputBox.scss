@use "../../styles/variables.scss" as *;
@use "../../styles/typography.scss" as *;

.input-box {
  &__label {
    font-family: $font-primary;
    color: $color-black;
    font-size: 0.75rem;
    font-weight: 500;
    line-height: 1.2rem;
    display: block;
  }

  &__input {
    height: 2.813rem;
    border: 0.063rem solid #e8e8e8;
    border-radius: 0.25rem;
    color: #000000;
    margin-top: 0.313rem;
    margin-bottom: 0.625rem;
    width: 22.125rem;
  }
}
