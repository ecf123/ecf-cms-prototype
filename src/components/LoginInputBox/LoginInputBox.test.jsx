import { customRender } from "../../utils/testUtils";
import LoginInputBox from "./LoginInputBox";
import { render, screen } from "@testing-library/react";



//snapshot test
it("Should render the login input box component", () => {
    const { container } = customRender(
      <LoginInputBox/>
    );
    expect(container).toMatchSnapshot();
  });

//test for text to appear?
it("should render the correct label", () => {
    render(<LoginInputBox labelText="label"/>)
    const labelText = screen.queryByText("label");
    expect(labelText).toBeInTheDocument();
})

it("should render the input box", () => {
    render(<LoginInputBox />)
    const inputBox = screen.getByAltText("inputbox");
    expect(inputBox).toBeInTheDocument();
})
