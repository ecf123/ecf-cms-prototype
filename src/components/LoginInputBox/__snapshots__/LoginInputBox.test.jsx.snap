// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render the login input box component 1`] = `
<div>
  <div
    class="input-box"
  >
    <label
      class="input-box__label"
    />
    <input
      alt="input box"
    />
  </div>
</div>
`;
